src/main/java/org/opensearch/security/dlic/rest/api/AbstractApiAction.java

```diff
@@ -558,8 +569,7 @@ public String getName() {
 	protected abstract Endpoint getEndpoint();
 
 	protected boolean isSuperAdmin() {
-		User user = threadPool.getThreadContext().getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
```

Comment:
The implementation for this function should not change. Super admin is reserved in the security plugin to indicate that a user is using client cert authentication and the principal of the cert matches a configured `admin_dn` in `opensearch.yml`.

Reply:
and it isn't change. I just move this check here: https://github.com/willyborankin/security/blob/5110fb1fb7a673177641a6803cbc16ea56b24ec4/src/main/java/org/opensearch/security/dlic/rest/api/RestApiAdminPrivilegesEvaluator.java#L98. I think it is better to keep this code in one place instead of:
```java
if (adminDNs.isAdmin(user.getName()) && ....)
``` 
otherwise tests wont pass for existing behave

Reply:
I believe the default configuration gives admin the permission to perform:

```
nodesdn
allowlist
actiongroup
user / internalusers
roles
rolesmapping
tenants
```

but not 

```
reloadcerts
```

Only the superadmin (connected via client cert) can perform `reloadcerts` now.

After this change I would still expect the `admin` to perform the same actions without additional config. 

Thank you for addressing all of the PR comments right away, this change looks good to me. Thank you for the contribution!